backend/src/schema/resolvers/moderation.spec.js

```diff
@@ -1,420 +1,349 @@
-            client.request('{ Comment(disabled: true) { id disabled } }'),
-          ).resolves.toEqual(expected)
-        })
-      })
+          it('returns disabled resource id', async () => {
+            variables = { id: 'comment-id' }
+            await expect(mutate({ mutation: disableMutation, variables })).resolves.toMatchObject({
+              data: { disable: 'comment-id' },
+              errors: undefined,
+            })
+          })
+
+          it('changes .disabledBy', async () => {
+            variables = { id: 'comment-id' }
+            const before = { data: { Comment: [{ id: 'comment-id', disabledBy: null }] } }
+            const expected = {
+              data: { Comment: [{ id: 'comment-id', disabledBy: { id: 'moderator-id' } }] },
+            }
+            await expect(query({ query: commentQuery, variables })).resolves.toMatchObject(before)
+            await expect(mutate({ mutation: disableMutation, variables })).resolves.toMatchObject({
+              data: { disable: 'comment-id' },
+            })
+            await expect(query({ query: commentQuery, variables })).resolves.toMatchObject(expected)
+          })
 
-      describe('on a post', () => {
-        beforeEach(async () => {
-          variables = {
-            id: 'p9',
-          }
+          it('updates .disabled on comment', async () => {
+            variables = { id: 'comment-id' }
+            const before = { data: { Comment: [{ id: 'comment-id', disabled: false }] } }
+            const expected = { data: { Comment: [{ id: 'comment-id', disabled: true }] } }
 
-          createResource = async () => {
-            await factory.create('User', { email: 'author@example.org', password: '1234' })
-            await factory.authenticateAs({ email: 'author@example.org', password: '1234' })
-            await factory.create('Post', {
-              id: 'p9', // that's the ID we will look for
-              categoryIds,
+            await expect(query({ query: commentQuery, variables })).resolves.toMatchObject(before)
+            await expect(mutate({ mutation: disableMutation, variables })).resolves.toMatchObject({
+              data: { disable: 'comment-id' },
             })
-          }
+            await expect(query({ query: commentQuery, variables })).resolves.toMatchObject(expected)
+          })
         })
 
-        it('returns disabled resource id', async () => {
-          const expected = { disable: 'p9' }
-          await setup()
-          await expect(action()).resolves.toEqual(expected)
-        })
+        describe('moderate a post', () => {
+          beforeEach(async () => {
+            variables = {}
+            await factory.create('Post', {
+              id: 'sample-post-id',
+            })
+          })
 
-        it('changes .disabledBy', async () => {
-          const before = { Post: [{ id: 'p9', disabledBy: null }] }
-          const expected = { Post: [{ id: 'p9', disabledBy: { id: 'u7' } }] }
-
```

Comment:
> [<img alt="roschaefer" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-09-18T08:47:44Z" title="Wednesday, September 18th 2019, 10:47:44 am +02:00">Sep 18, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -54,369 +63,369 @@ describe('disable', () => {
   beforeEach(() => {
     // our defaul set of variables
     variables = {
-      id: 'blabla',
+      id: 'some-resource',
     }
   })
-
-  const action = async () => {
-    return client.request(mutation, variables)
-  }
-
-  it('throws authorization error', async () => {
-    await setup()
-    await expect(action()).rejects.toThrow('Not Authorised')
-  })
-
-  describe('authenticated', () => {
-    beforeEach(() => {
-      authenticateClient = setupAuthenticateClient({
-        email: 'user@example.org',
-        password: '1234',
-      })
+  describe('unauthenticated', () => {
+    beforeEach(async () => {
+      authenticatedUser = await currentUser.toJson()
     })
-
     it('throws authorization error', async () => {
-      await setup()
-      await expect(action()).rejects.toThrow('Not Authorised')
+      const { errors } = await mutate({ mutation, variables })
+      expect(errors[0]).toHaveProperty('message', 'Not Authorised!')
     })
+  })
+  describe('authenticated', () => {
+    describe('i am not a moderator', () => {
+      let nonModerator
 
-    describe('as moderator', () => {
-      beforeEach(() => {
-        authenticateClient = setupAuthenticateClient({
-          id: 'u7',
-          email: 'moderator@example.org',
+      beforeEach(async () => {
+        nonModerator = await factory.create('User', {
+          id: 'non-moderator',
+          name: 'Non Moderator',
+          email: 'non.moderator@example.org',
           password: '1234',
-          role: 'moderator',
-        })
-      })
-
-      describe('on something that is not a (Comment|Post|User) ', () => {
-        beforeEach(async () => {
-          variables = {
-            id: 't23',
-          }
-          createResource = () => {
-            return Promise.all([factory.create('Tag', { id: 't23' })])
-          }
-        })
-
-        it('returns null', async () => {
-          const expected = { disable: null }
-          await setup()
-          await expect(action()).resolves.toEqual(expected)
-        })
-      })
-
-      describe('on a comment', () => {
-        beforeEach(async () => {
-          variables = {
-            id: 'c47',
-          }
-          createPostVariables = {
-            id: 'p3',
-            title: 'post to comment on',
-            content: 'please comment on me',
-            categoryIds,
-          }
-          createCommentVariables = {
-            id: 'c47',
-            postId: 'p3',
-            content: 'this comment was created for this post',
-          }
-          createResource = async () => {
-            await factory.create('User', {
-              id: 'u45',
-              email: 'commenter@example.org',
-              password: '1234',
-            })
-            const asAuthenticatedUser = await factory.authenticateAs({
-              email: 'commenter@example.org',
-              password: '1234',
-            })
-            await asAuthenticatedUser.create('Post', createPostVariables)
-            await asAuthenticatedUser.create('Comment', createCommentVariables)
-          }
-        })
-
-        it('returns disabled resource id', async () => {
-          const expected = { disable: 'c47' }
-          await setup()
-          await expect(action()).resolves.toEqual(expected)
-        })
-
-        it('changes .disabledBy', async () => {
-          const before = { Comment: [{ id: 'c47', disabledBy: null }] }
-          const expected = { Comment: [{ id: 'c47', disabledBy: { id: 'u7' } }] }
-
-          await setup()
-          await expect(client.request('{ Comment { id,  disabledBy { id } } }')).resolves.toEqual(
-            before,
-          )
-          await action()
-          await expect(
-            client.request('{ Comment(disabled: true) { id,  disabledBy { id } } }'),
-          ).resolves.toEqual(expected)
-        })
-
-        it('updates .disabled on comment', async () => {
-          const before = { Comment: [{ id: 'c47', disabled: false }] }
-          const expected = { Comment: [{ id: 'c47', disabled: true }] }
-
-          await setup()
-          await expect(client.request('{ Comment { id disabled } }')).resolves.toEqual(before)
-          await action()
-          await expect(
-            client.request('{ Comment(disabled: true) { id disabled } }'),
-          ).resolves.toEqual(expected)
-        })
-      })
-
-      describe('on a post', () => {
-        beforeEach(async () => {
-          variables = {
-            id: 'p9',
-          }
-
-          createResource = async () => {
-            await factory.create('User', { email: 'author@example.org', password: '1234' })
-            await factory.authenticateAs({ email: 'author@example.org', password: '1234' })
-            await factory.create('Post', {
-              id: 'p9', // that's the ID we will look for
-              categoryIds,
-            })
-          }
-        })
-
-        it('returns disabled resource id', async () => {
-          const expected = { disable: 'p9' }
-          await setup()
-          await expect(action()).resolves.toEqual(expected)
-        })
-
-        it('changes .disabledBy', async () => {
-          const before = { Post: [{ id: 'p9', disabledBy: null }] }
-          const expected = { Post: [{ id: 'p9', disabledBy: { id: 'u7' } }] }
-
-          await setup()
-          await expect(client.request('{ Post { id,  disabledBy { id } } }')).resolves.toEqual(
-            before,
-          )
-          await action()
-          await expect(
-            client.request('{ Post(disabled: true) { id,  disabledBy { id } } }'),
-          ).resolves.toEqual(expected)
-        })
-
-        it('updates .disabled on post', async () => {
-          const before = { Post: [{ id: 'p9', disabled: false }] }
-          const expected = { Post: [{ id: 'p9', disabled: true }] }
-
-          await setup()
-          await expect(client.request('{ Post { id disabled } }')).resolves.toEqual(before)
-          await action()
-          await expect(client.request('{ Post(disabled: true) { id disabled } }')).resolves.toEqual(
-            expected,
-          )
         })
+        authenticatedUser = await nonModerator.toJson()
       })
-    })
-  })
-})
-
-describe('enable', () => {
-  const mutation = gql`
-    mutation($id: ID!) {
-      enable(id: $id)
-    }
-  `
-  let variables
-
-  const action = async () => {
-    return client.request(mutation, variables)
-  }
-
-  beforeEach(() => {
-    // our defaul set of variables
-    variables = {
-      id: 'blabla',
-    }
-  })
-
-  it('throws authorization error', async () => {
-    await setup()
-    await expect(action()).rejects.toThrow('Not Authorised')
-  })
-
-  describe('authenticated', () => {
-    beforeEach(() => {
-      authenticateClient = setupAuthenticateClient({
-        email: 'user@example.org',
-        password: '1234',
+      it('throws authorization error', async () => {
+        const { errors } = await mutate({ mutation, variables })
+        expect(errors[0]).toHaveProperty('message', 'Not Authorised!')
       })
     })
 
-    it('throws authorization error', async () => {
-      await setup()
-      await expect(action()).rejects.toThrow('Not Authorised')
-    })
-
-    describe('as moderator', () => {
+    describe('I am a moderator', () => {
+      let moderator
       beforeEach(async () => {
-        authenticateClient = setupAuthenticateClient({
-          role: 'moderator',
-          email: 'someuser@example.org',
+        moderator = await factory.create('User', {
+          id: 'moderator-id',
+          name: 'Moderator',
+          email: 'moderator@example.org',
           password: '1234',
+          role: 'moderator',
         })
+        authenticatedUser = await moderator.toJson()
       })
 
-      describe('on something that is not a (Comment|Post|User) ', () => {
+      describe('moderate a resource that is not a (Comment|Post|User) ', () => {
         beforeEach(async () => {
           variables = {
-            id: 't23',
-          }
-          createResource = () => {
-            // we cannot create a :DISABLED relationship here
-            return Promise.all([factory.create('Tag', { id: 't23' })])
+            id: 'sample-tag-id',
           }
+          await Promise.all([factory.create('Tag', { id: 'sample-tag-id' })])
         })
 
         it('returns null', async () => {
-          const expected = { enable: null }
-          await setup()
-          await expect(action()).resolves.toEqual(expected)
+          const expected = { data: { disable: null } }
+          await expect(mutate({ mutation, variables })).resolves.toMatchObject(expected)
```


Not too much code here, so you could merge these two lines,

Reply:
> [<img alt="aonomike" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/aonomike) **Authored by [aonomike](https://github.com/aonomike)**
_<time datetime="2019-09-18T19:28:00Z" title="Wednesday, September 18th 2019, 9:28:00 pm +02:00">Sep 18, 2019</time>_

---


Done